sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/ReduceNumShufflePartitions.scala

```diff
@@ -182,13 +182,13 @@ case class ReduceNumShufflePartitions(conf: SQLConf) extends Rule[SparkPlan] {
 }
 
 case class CoalescedShuffleReaderExec(
-    child: QueryStageExec,
+    // `child` is usually `ShuffleQueryStageExec` or `ReusedQueryStageExec`, but can be the shuffle
+    // exchange node during canonicalization.
```

Comment:
https://github.com/apache/spark/blob/master/sql/core/src/main/scala/org/apache/spark/sql/execution/SortExec.scala#L32-L39

We can add descriptions for this case class. It will look more formal.